### PR TITLE
fix: Use a proper macro to handle libbpf error returns

### PR DESCRIPTION
**Pull request by simar7**
Fixes: https://github.com/aquasecurity/tracee/issues/691

Signed-off-by: Simarpreet Singh <simar@linux.com>

### CODE REVIEW

**Comment by simar7** [comment]
@yanivagman Any ideas how we can test this?

File: libbpfgo/libbpfgo.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ struct perf_buffer * init_perf_buf(int map_fd, int page_cnt, uintptr_t ctx) {
     pb_opts.lost_cb = perfLostCallback;
     pb_opts.ctx = (void*)ctx;
     pb = perf_buffer__new(map_fd, page_cnt, &pb_opts);
-    if (pb < 0) {
+    if IS_ERR_VALUE(pb) {
```
Comment (yanivagman) [inline libbpfgo/libbpfgo.go:55]: We should probably do the same for init_ring_buf

**Review by yanivagman — COMMENTED** [review]
Shouldn't we include the err.h header explicitly? I'm not sure how this even works

**Comment by yanivagman** [comment]
About how to test - I believe that we can just give some invalid argument.
For example, invalid map_fd should trigger an error

**Comment by simar7** [comment]
> About how to test - I believe that we can just give some invalid argument.
> For example, invalid map_fd should trigger an error

I tried this but wasn't successful. I mentioned earlier in this comment https://github.com/aquasecurity/tracee/issues/691#issuecomment-816253557 

Any ideas what am I missing?

**Comment by simar7** [comment]
> About how to test - I believe that we can just give some invalid argument.
> For example, invalid map_fd should trigger an error

Maybe I'm doing something wrong here but I wasn't able to reproduce this on my machine. https://github.com/aquasecurity/tracee/issues/691#issuecomment-816253557

Any idea what am I doing differently?

**Comment by yanivagman** [comment]
You are probably not using your local copy of libbpfgo, but from the repo.
Did you edit go.mod of tracee-ebpf to use it?
`replace github.com/aquasecurity/tracee/libbpfgo => ../libbpfgo`

I tested with a negative map_fd given to init_perf_buf and it worked (meaning failed with an error :-) )

**Comment by simar7** [comment]
> You are probably not using your local copy of libbpfgo, but from the repo.
> Did you edit go.mod of tracee-ebpf to use it?
> `replace github.com/aquasecurity/tracee/libbpfgo => ../libbpfgo`
> 
> I tested with a negative map_fd given to init_perf_buf and it worked (meaning failed with an error :-) )

Thanks that was it. I tested with a negative map_fd and it did work.

File: libbpfgo/libbpfgo.go [hunk 2]
```diff
@@ -23,6 +23,14 @@ extern void perfLostCallback(void *ctx, int cpu, __u64 cnt);
 
 extern int ringbufferCallback(void *ctx, void *data, size_t size);
 
+#define MAX_ERRNO       4095
```
Comment (yanivagman) [inline libbpfgo/libbpfgo.go:26]: Can you please add a comment here that this was taken from err.h of libbpf?

Comment (simar7) [reply libbpfgo/libbpfgo.go:38]: Yes https://github.com/aquasecurity/tracee/pull/694/commits/1f4717ba14be975d4efe2aee3ee9f84b25aade57#diff-4108726d54769f59926ec61abdeaab04a651cdf8766a24a74a859b4d98befbc8R37

File: libbpfgo/libbpfgo.go [hunk 3]
```diff
@@ -1,5 +1,16 @@
 package libbpfgo
 
+import "C"
```
Comment (yanivagman) [inline libbpfgo/libbpfgo.go:3]: Any reason to move all the imports here?
Import "c" should be after the c code (and from all the cgo examples I saw, all the other imports come after it)

**Comment by simar7** [comment]
> @simar7 can we close this PR in favor of #709?

Yeah good idea.

**Pull request closed without merging**